lib/src/sdk_env.dart

```diff
@@ -267,6 +267,20 @@ class PubEnvironment {
       );
     }
   }
+
+  /// Removes the dev_dependencies from the pubspec.yaml
+  Future removeDevDependencies(String packageDir) async {
+    final file = new File(p.join(packageDir, 'pubspec.yaml'));
+    final original = await file.readAsString();
+    final replaced = original
+        .split('\n')
+        .map((line) => line.trim().startsWith('dev_dependencies')
+            ? line.replaceFirst(
+                'dev_dependencies', '__pub_replaced__dev_dependencies')
```

Comment:
This really scares me. JSON is valid yaml. You could just read in the Yaml as a map, remove `dev_dependencies` and then write it back as JSON. Should work fine – and be less buggy

Reply:
right, fixed it